Feat: support for puppeteer-extra plugins

xtrime-ru opened the pull request:
Corresponding issue: https://github.com/rialto-php/puphpeteer/issues/67

Added new option to allow override puppeteer instance.

This is example usage of [puppeteer-extra-plugin-stealth](https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth): 

```php
<?php
require "vendor/autoload.php";

use Nesk\Puphpeteer\Puppeteer;

$puppeteer = new Puppeteer([
    'js_extra' => /** @lang JavaScript */ "
        const puppeteer = require('puppeteer-extra');
        const StealthPlugin = require('puppeteer-extra-plugin-stealth');
        puppeteer.use(StealthPlugin());
        instruction.setDefaultResource(puppeteer);
    ",
]);
$browser = $puppeteer->launch();

$page = $browser->newPage();

$page->goto('https://bot.sannysoft.com');
$page->waitForTimeout(5000);
$page->screenshot(['path' => 'example2.png', 'fullPage'=> true]);

$browser->close();
```

Without plugin: 
![example2](https://user-images.githubusercontent.com/34161928/129078971-1346cfa4-cc82-4cbc-b0f1-53a4e86ef54f.png)

With plugin (code above): 
![example](https://user-images.githubusercontent.com/34161928/129078940-bdeb8dbb-3447-4494-b776-986fa465eb19.png)

xtrime-ru committed:
Feat: support for puppeteer-extra plugins

Dr10s commented:
I'm waiting for this feature <3

btidude commented:
I'm also waiting for this feature

overgapo commented:
👍

jsandeo commented:
👍

jfrancuz commented:
@xtrime-ru - thanks a lot!
@nesk, this feature is very much needed - please review the PR. Hopefully, it can be accepted!

xtrime-ru commented:
> @xtrime-ru - thanks a lot!

You welcome! While waiting nesk to aprove my PR you can use [our fork](https://github.com/rialto-php/puphpeteer/compare/dev...zoonru:puppeteer-10) with all new features: 

- Updated and fixed PHPDOC (PhpStorm compatible)
- Puppeteer ^13 support ( its little outdated now, but better than nothing )
- puppeteer-extra-plugin-stealth support

TheWildHorse commented:
Wow! Nice job dude! Would love for this to get this merged.
